File: Chapter 2-14-Check Version App Store/Objective-C/CheckVersion/CheckVersion/ViewController.m
//
//  ViewController.m
//  CheckVersion
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  
  [self getAppVersionAndRunOnCompletion:^(NSString *version) {
    NSLog(@"latest version on the app store = %@", version);
  }];
  
}


-(void)getAppVersionAndRunOnCompletion:(void (^)(NSString *version))completionBlock {
  
  // get the app’s bundle identifier
  NSString *bundleIdentifier = @"com.yourCompany.yourAppBundleID";
  
  // use iTunes API to get data about the application
  NSURL *lookupURL = [NSURL URLWithString:[NSString stringWithFormat:
                                           @"http://itunes.apple.com/lookup?bundleId=%@",
                                           bundleIdentifier]];
  
  
  NSURLSession *session = [NSURLSession sharedSession];
  [[session dataTaskWithURL:lookupURL
          completionHandler:^(NSData *data,
                              NSURLResponse *response,
                              NSError *error) {
            
            if ( (data == nil) || (error != nil) ) {
              completionBlock(nil);
              return;
            }
            
            NSDictionary *jsonResults = [NSJSONSerialization
                                         JSONObjectWithData:data options:0 error:nil];
            
            NSUInteger resultCount = [[jsonResults objectForKey:
                                       @"resultCount"] integerValue];
            
            // if number of objects on jsonResults is zero, exit with nil
            if (!resultCount){
              completionBlock(nil);
              return;
            }
            
            NSDictionary *appDetails = [[jsonResults objectForKey:
                                         @"results"] firstObject];
            NSString *latestVersion = [appDetails objectForKey:
                                       @"version"];
            
            completionBlock(latestVersion);

            
          }] resume];
  
}



- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: Chapter 2-14-Check Version App Store/Swift 2.0/CheckVersion/CheckVersion/ViewController.swift
//
//  ViewController.swift
//  CheckVersion
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    
    getAppVersion { (version) -> Void in
      print("latest version = ", version)
    }
    
  }
  
  
  func getAppVersion ( onCompletion: (version: String) -> Void) {
    
    // get the app’s bundle identifier
    let bundleIdentifier = "com.addfone.EuroMillionsIPAD"
    
    // build the app path using iTunes API
    let path : String = "http://itunes.apple.com/lookup?bundleId=\(bundleIdentifier)"
    
    
    let lookupURL : NSURL = NSURL(string:path)!
    let session = NSURLSession.sharedSession()
    
    let request = NSURLRequest(URL: lookupURL)
    
    // request data about the application, asynchronously
    // using iTunes API
    let task = session.dataTaskWithRequest(request, completionHandler: { data, response, error in
      if !(error != nil) {
      do {
      let jsonResults = try  NSJSONSerialization.JSONObjectWithData(data!, options: [])  as! NSDictionary
      
      let results = jsonResults["results"]
      let appDetails = results?.firstObject
      
      // get the application version
      let latestVersion = appDetails!!["version"] as! String
      onCompletion(version: latestVersion)
    }
      catch let error as NSError {
      print("error = \(error)")
    }
      catch {   }
      }
      })
    
    task.resume()
    
  }
  
  
  
  
  

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }


}

